### PR TITLE
teos: 0.1.2 -> 0.2.0

### PR DESCRIPTION
**Pull request by seberm**
###### Description of changes

Update teos package to v0.2.0:
- https://github.com/talaia-labs/rust-teos/releases/tag/v0.2.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/teos/default.nix [hunk 1]
```diff
@@ -28,7 +29,7 @@ let
     platforms = platforms.unix;
   };
 
-  cargoPatches = [ ./add-cargo-lock.patch ];
+  cargoPatches = [];
```
Comment (prusnak) [inline pkgs/applications/blockchains/teos/default.nix:32]: We can drop this line completely.

(and drop `cargoPatches` from `inherit` two times below)

**Review by prusnak — CHANGES_REQUESTED** [review]
`cargoSha256` of `teos-watchtower-plugin` needs to be changed too (to `sha256-3ke1qTFw/4I5dPLuPjIGp1n2C/eRfPB7A6ErMFfwUzE=`)

File: pkgs/applications/blockchains/teos/default.nix [hunk 2]
```diff
@@ -60,10 +60,10 @@ in
 
   teos-watchtower-plugin = rustPlatform.buildRustPackage {
     pname = "teos-watchtower-plugin";
-    cargoSha256 = "sha256-xL+DiEfgBYJQ1UJm7LAr1/f34pkU8FRl4Seic8MFAlM=";
+    cargoSha256 = "sha256-3ke1qTFw/4I5dPLuPjIGp1n2C/eRfPB7A6ErMFfwUzE=)";
```
Comment (prusnak) [inline pkgs/applications/blockchains/teos/default.nix:63]: typo:
```suggestion
    cargoSha256 = "sha256-3ke1qTFw/4I5dPLuPjIGp1n2C/eRfPB7A6ErMFfwUzE=";
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by prusnak** [comment]
@GrahamcOfBorg build teos teos-watchtower-plugin

**Comment by prusnak** [comment]
It seems these tests fail on Linux now (both aarch64 and x86_64):

```
retrier::tests::test_manage_retry_misbehaving
retrier::tests::test_manage_retry_reachable
retrier::tests::test_manage_retry_rejected
retrier::tests::test_manage_retry_subscription_error
retrier::tests::test_manage_retry_while_idle
```

**Comment by prusnak** [comment]
On aarch64-darwin I see failure on

```
retrier::tests::test_retry_tower_rejected
retrier::tests::test_retry_tower_subscription_error
```

----

x86_64-darwin works just fine

**Comment by prusnak** [comment]
So maybe the following patch disables the tests we need to disable?

``` patch
diff --git a/pkgs/applications/blockchains/teos/default.nix b/pkgs/applications/blockchains/teos/default.nix
index f031ffdb32e..1f9ff4b807b 100644
--- a/pkgs/applications/blockchains/teos/default.nix
+++ b/pkgs/applications/blockchains/teos/default.nix
@@ -69,9 +69,17 @@ in
       description = "A Lightning watchtower plugin for clightning";
     };
 
-    # The test is skipped due to following error:
-    #   thread 'retrier::tests::test_manage_retry_unreachable' panicked at 'assertion failed:
-    #   wt_client.lock().unwrap().towers.get(&tower_id).unwrap().status.is_unreachable()', watchtower-plugin/src/retrier.rs:518:9
-    checkFlags = lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [ "--skip=retrier::tests::test_manage_retry_unreachable" ];
+    checkFlags = lib.optionals stdenv.isLinux [
+      "--skip=retrier::tests::test_manage_retry_misbehaving"
+      "--skip=retrier::tests::test_manage_retry_reachable"
+      "--skip=retrier::tests::test_manage_retry_rejected"
+      "--skip=retrier::tests::test_manage_retry_subscription_error"
+    ] ++ lib.optionals (stdenv.isLinux && stdenv.isAarch64) [
+      "--skip=retrier::tests::test_manage_retry_while_idle"
+    ] ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [
+      "--skip=retrier::tests::test_manage_retry_unreachable"
+      "--skip=retrier::tests::test_retry_tower_rejected"
+      "--skip=retrier::tests::test_retry_tower_subscription_error"
+    ];
   };
 }
```

**Comment by prusnak** [comment]
@GrahamcOfBorg build teos teos-watchtower-plugin

**Comment by figsoda** [comment]
opened #219088 as an alternative

**Pull request closed without merging**